### PR TITLE
Cleanup Intel driver & remove some hard coded threads = 32

### PR DESCRIPTION
**Pull request by etiotto**

### CODE REVIEW

**Review by whitneywhtsang — APPROVED** [review]

**Comment by pbchekin** [comment]
Lit tests stuck at
```
PASS: TRITON :: TritonGPU/optimize-locality.mlir (25 of 39)
```
in every attempt.

File: include/triton/Dialect/TritonGPU/IR/TritonGPUDialect.td [hunk 1]
```diff
@@ -46,7 +46,7 @@ def TritonGPU_Dialect : Dialect {
     static int getThreadsPerWarp(ModuleOp mod) {
       Attribute threadsPerWarp = mod->getDiscardableAttr("triton_gpu.threads-per-warp");
       if(!threadsPerWarp) {
-        return 32;
+        return triton::gpu::TritonGPUDialect::getThreadsPerWarp(mod);
```
Comment (chengjunlu) [inline include/triton/Dialect/TritonGPU/IR/TritonGPUDialect.td:49]: It goes endless nested recursive calling `getThreadsPerWarp` -> `getThreadsPerWarp`.

The default number of 32 is for the case that the old `ttgir` without the `threads-per-warp` attribute.

Comment (etiotto) [reply include/triton/Dialect/TritonGPU/IR/TritonGPUDialect.td:49]: I see. We will need to find a way to make this code generic and use 16 for XPU and 32 for NV (and possibly 64 for AMD).

Comment (etiotto) [reply include/triton/Dialect/TritonGPU/IR/TritonGPUDialect.td:49]: Fir this PR I will leave the original code.

**Pull request merged**